packages/fluentui/react-northstar/src/types.ts

```diff
@@ -64,7 +64,9 @@ type ReactNode = React.ReactChild | React.ReactNodeArray | React.ReactPortal | b
 
 export type ShorthandValue<P extends Props> = ReactNode | ObjectShorthandValue<P>;
 
-export type ShorthandCollection<P, K = never> = ShorthandValue<P & { kind?: K }>[];
+export type ShorthandCollection<P, K = never> = ShorthandValue<
+  P & { kind?: keyof K } & { [Key in keyof K[keyof K]]?: K[keyof K][Key] }
+>[];
```

Comment:
`{ [Key in keyof K[keyof K]]?: K[keyof K][Key] }` not sure that I understand this part

- `K` is `{ item: Foo, divider: Bar }`
- `keyof K` => `item | divider`
- `keyof K[keyof K]` => ???

I don't think that it's expected result:
![image](https://user-images.githubusercontent.com/14183168/81539738-f7215b00-9370-11ea-8ffa-4666ea190ce6.png)

https://www.typescriptlang.org/play?#code/C4TwDgpgBA0lC8UDeAoK6oEtgQLYC4oBnYAJ0wDsBzNDAE0wDdM6JTCKBXXAIzZQC+KFKEiwAjAigBrCCAD2AM1gjw0GACYpshcpjigA

Reply:
`keyof K[keyof K]` is the keys in `Foo` or `Bar`

Reply:
I still don't understand:

```tsx
type ItemProps = {
    yellow: boolean
    orange: boolean
}

type DividerProps = {
    blue: boolean
    green: boolean
}

type Kinds = {
    item: ItemProps
    divider: DividerProps
}
```

=>

```
// { [Key in keyof K[keyof K]]?: K[keyof K][Key] }
=>
// keyof K is `item` | `divider`
// keyof K[keyof K] is what? `keyof Kinds['item']` is `yellow | orange`
// So if `Key` is `yellow | orange` will add what?
```

Currently it works only because `Props` adding `any`
[Broken playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAksEFsAKAnA9mAzlAvFA3gFBQlQgQA2FaA7gFxQBGaaFEAhgHbGlopcBzCA2asO3AL6FCoSFAAiASwBuigCYQUqDNjxFSTCgFdhTFmy48SAlBAicR58YSkzw0ANKLOa3QStQivAIDHCI2lgBairqmgxKqhpa6JGustAAygAWfMBZXGoAwqxsAMbAimicADxIADRQHrhQnBDKmgB8zdm5+T4AauzGENU8SFAAZARQANbeagD8DDMQIGgAZo1QEpPTANoeq4Gcs6sbjXsra5seALq3SxdX53cHq7fbhB17twDc0ukoBFMNUACrNfASLp4cEAelhuwAShBSnw1NVMMAUN4BA0uCAOv9AQB5RgAKxRwB6KDyBUGw1qUAgAA94D5sMDoUCUiCkIS3HJqbSBkMTIyWWzfNydFzSRTykK+mp6WK+f8BdAAILNZ43eaYPYAciCiENtwB7igoOB3RyNKVxSolMqNXwUHWLAYmOxnAE2waXnZHWkqM4mKg7AY1p5zT2ATdcx8DEN0USmkNDQ9aGT7AzhhMDCxJk+tyAA)